### ISSUE TITLE
Incompatibility with Sigma Dimensions

### ISSUE THREAD

**Issue by pap1723** [jrossignol/ContractConfigurator#626]
Hey @jrossignol 
Most of the contracts that I have written are using the dynamic information gathering that you have put into CC. So instead of a contract saying reach 140,000 m, it says, to reach HomeWorld().AtmosphereAltitude() so that it works in stock as well as scaled systems.

Apparently there is an issue with Sigma Dimensions where the contract is pulling the data for the planetary bodies before the rescale is happening. So on Stock Size RSS, all of the contracts are reading the small planet size as a full size Earth.

Let me know if you have any questions or need any documentation to make it clearer.

Thanks,
Pap

**Comment by jrossignol** [comment]
I'd have to dig a bit deeper, but I'm surprised there's an issue here.  I
thought that SD uses Kopernicus under the hood, which runs on the prefabs
well before contrac Contract Configurator runs (on main menu).  Maybe
@Sigma88 can comment if there's an obvious piece I'm missing there.

As far as a workaround, changing the expression to be non-deterministic
will force it to be reevaluated each time a contract is generated (by which
time the value will be correct).  One way would be to do something like
this:

[ HomeWorld().AtmosphereAltitude() ].Random()


On May 8, 2017 6:58 AM, "Pap" <notifications@github.com> wrote:

Hey @jrossignol <https://github.com/jrossignol>
Most of the contracts that I have written are using the dynamic information
gathering that you have put into CC. So instead of a contract saying reach
140,000 m, it says, to reach HomeWorld().AtmosphereAltitude() so that it
works in stock as well as scaled systems.

Apparently there is an issue with Sigma Dimensions where the contract is
pulling the data for the planetary bodies before the rescale is happening.
So on Stock Size RSS, all of the contracts are reading the small planet
size as a full size Earth.

Let me know if you have any questions or need any documentation to make it
clearer.

Thanks,
Pap

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/jrossignol/ContractConfigurator/issues/626>, or mute
the thread

**Comment by Sigma88** [comment]
@jrossignol most of SD is config based but I've been able to add a plugin to do some mathemagic that is not allowed on MM

specifically, I think the problem here is with the feature "atmoTopLayer"

I'll explaina bit how the feature works to give you context, if you don't care skip to the TLDR

basically atmosphere scaling now is handled by 2 parameters, *Atmosphere* and *atmoTopLayer*

*Atmosphere* scales the whole curve so that the maxAltitude is multiplied by the parameter itself

*atmoTopLayer* on the contrary extrapolates the shape of the curve, and extends it or trims it, this was required to give bigger atmospheres a more gradual fade-out to 0 pression, and to avoid having big layers of 0.000something pressure on smaller atmospheres

the *Atmosphere* parameter is processed by the cfgs and the atmoTopLayer is processed by the plugin


TLDR:
I think your mod might be reading the values before mine edits them, [this is probably the relevant code](https://github.com/Sigma88/Sigma-Dimensions/blob/master/%5BSource%5D/SigmaDimensions/AtmosphereTopLayer.cs) which runs at *KSPAddon.Startup.MainMenu* - *Start()*

**Comment by jrossignol** [comment]
Fixed for next release.
